### ISSUE TITLE
[BUG] - jpg and webp formats don't store the ComfyUI workflow

### ISSUE THREAD

**Issue by Dachiko** [receyuki/comfyui-prompt-reader-node#57]
### Description

Any format except PNG doesn't contain or contain broken workflow. Nothing happens upon dragging the picture to the browser window.

### Image file

![06173](https://github.com/receyuki/comfyui-prompt-reader-node/assets/1128495/fd3c8f82-b612-4041-a0e3-43e0215fd251)
![image](https://github.com/receyuki/comfyui-prompt-reader-node/assets/1128495/d356f16d-8810-4123-8c35-ab3c3e7077fc)

**Comment by receyuki** [comment]
You are right. But the problem is the built-in saver doesn't support formats other than PNG, so I don't know what the JPG or WebP metadata readable by comfyui looks like. I will look into how it is handled in other nodes, but I still recommend using PNG as the main output format, or saving in both PNG and other formats simultaneously.

**Comment by Dachiko** [comment]
Thanks! PNGs are all right, it just bugs me that I can have same images to take 3x less space. Other nodes can store the workflow, but I love to use your SD Prompt Reader, so yeah, sticking to png for now.

**Comment by receyuki** [comment]
I have checked all the savers I know, and among the formats they support, only WebP and PNG include workflow. I'll definitely fix the support for WebP. But for JPG, could you try again, confirm that the workflow can be embedded into JPG, and then let me know the name of the node set?

**Comment by Dachiko** [comment]
Thanks! I don't know if any other extensions saves workflow into JPG. It was long time ago, and I only remember I tried some, and WebP stored workflows for sure. I was under impression that any format listed in the node should work the same, so though it was a bug. If you'll implement embedding workflow into WebP, I'll be happy. But maybe there should be a warning message somewhere telling that JPG is a bit crippled, so that people won't lost their workflows without knowing it.

**Comment by receyuki** [comment]
Bad news: I originally planned to use the WAS's method to write the workflow into webp, but after some tests I found that the workflow and metadata cannot coexist. Which means writing metadata into webp will cause its workflow to be unreadable by ComfyUI. Therefore, I need to give up this feature.

**Comment by Dachiko** [comment]
F
Thanks for trying!
